app/src/main/java/com/karumi/ui/presenter/SuperHeroDetailPresenter.kt

```diff
@@ -25,20 +28,25 @@ class SuperHeroDetailPresenter(
 
     override fun update() {
         view?.showLoading()
-        refreshSuperHeroes()
+        refreshSuperHero()
     }
 
-    private fun refreshSuperHeroes() = async {
+    private fun refreshSuperHero() = async {
         val result = await { getSuperHeroByName(name) }
         view?.hideLoading()
-        view?.showSuperHero(result)
+        when (result) {
+            is Right -> view?.showSuperHero(result.r)
+            is Left -> view?.showError(result.l)
+        }
+
     }
 
     interface View {
         fun close()
         fun showLoading()
         fun hideLoading()
         fun showSuperHero(superHero: SuperHero)
+        fun showError(l: DomainError)
```

Comment:
I shouldn't call this parameter `l` but `error`, same thing in the other `View` interface located in the `SuperHeroesPresenter` class

Reply:
this is a mistake, l is really bad name